cpp/rrtmgp_conversion.h

```diff
@@ -0,0 +1,956 @@
+
+template <typename YArray,
+          typename std::enable_if<!is_view_v<YArray>>::type* = nullptr>
+void p4d(const YArray& array, const std::string& name, int idx1, int idx2, int idx3, int idx4)
+{
+  const int adjust_val = std::is_same<typename YArray::non_const_value_type, int>::value ? 1 : 0;
+  std::cout << "JGFY " << name << "(" << idx1-1 << ", " << idx2-1 << ", " << idx3-1 << ", " << idx4-1 << ") = " << array(idx1, idx2, idx3, idx4) - adjust_val << std::endl;
+}
+
+template <typename KView,
+          typename std::enable_if<is_view_v<KView>>::type* = nullptr>
+void p5d(const KView& view, const std::string& name, int idx1, int idx2, int idx3, int idx4, int idx5)
+{ std::cout << "JGFK " << name << "(" << idx1 << ", " << idx2 << ", " << idx3 << ", " << idx4 << ", " << idx5 << ") = " << view(idx1, idx2, idx3, idx4, idx5) << std::endl; }
+
+template <typename YArray,
+          typename std::enable_if<!is_view_v<YArray>>::type* = nullptr>
+void p5d(const YArray& array, const std::string& name, int idx1, int idx2, int idx3, int idx4, int idx5)
+{
+  const int adjust_val = std::is_same<typename YArray::non_const_value_type, int>::value ? 1 : 0;
+  std::cout << "JGFY " << name << "(" << idx1-1 << ", " << idx2-1 << ", " << idx3-1 << ", " << idx4-1 << ", " << idx5-1 << ") = " << array(idx1, idx2, idx3, idx4, idx5) - adjust_val << std::endl;
+}
+
+template <typename KView,
+          typename std::enable_if<is_view_v<KView>>::type* = nullptr>
+void p6d(const KView& view, const std::string& name, int idx1, int idx2, int idx3, int idx4, int idx5, int idx6)
+{ std::cout << "JGFK " << name << "(" << idx1 << ", " << idx2 << ", " << idx3 << ", " << idx4 << ", " << idx5 << ", " << idx6 << ") = " << view(idx1, idx2, idx3, idx4, idx5, idx6) << std::endl; }
+
+template <typename YArray,
+          typename std::enable_if<!is_view_v<YArray>>::type* = nullptr>
+void p6d(const YArray& array, const std::string& name, int idx1, int idx2, int idx3, int idx4, int idx5, int idx6)
+{
+  const int adjust_val = std::is_same<typename YArray::non_const_value_type, int>::value ? 1 : 0;
+  std::cout << "JGFY " << name << "(" << idx1-1 << ", " << idx2-1 << ", " << idx3-1 << ", " << idx4-1 << ", " << idx5-1 << ", " << idx6-1 << ") = " << array(idx1, idx2, idx3, idx4, idx5, idx6) - adjust_val << std::endl;
+}
+
+// Copied from EKAT
+#define IMPL_THROW_RRT(condition, msg, exception_type)    \
+do {                                                      \
+  if ( ! (condition) ) {                                  \
+    std::stringstream _ss_;                               \
+    _ss_ << "\n FAIL:\n" << #condition  << "\n";          \
+    _ss_ << "\n" << msg;                                  \
+    throw exception_type(_ss_.str());                     \
+  }                                                       \
+} while(0)
+
+#define RRT_REQUIRE(condition, msg) IMPL_THROW_RRT(condition, msg, std::runtime_error)
```

Comment:
In general it would be nice to have a little more documentation in this header file for the functions it implements to explain what everything is doing. Most are self explanatory enough though I guess.

Reply:
Good point, I will add some.